src/transformStateWithClones.js

```diff
@@ -7,7 +7,52 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesHistory = [];
+  let newState = state;
+
+  for (const action of actions) {
+    newState = { ...newState };
+
+    switch (action.type) {
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. Good usage of the switch statement here.